eng/tools/sdk-suppressions/src/updateSdkSuppressionsLabel.ts

```diff
@@ -0,0 +1,318 @@
+    };
+    base: {
+        owner: string,
+        repo: string,
+        ref: string
+    };
+}
+
+/**
+ * 
+ * @param appClient 
+ * @param pr 
+ * @returns SdkName list
+ * This part compares the suppression files of the head branch and the base branch. 
+ * To get the SDK, we need to identify which package name, SDK name, or breaking changes are different and apply the SDK suppression label accordingly in the next step.
+ * change details can see at getSdkNamesWithChangedSuppressions function
+ * on the other hand that the sdkName list will return an empty array if it does not have a suppression file or if the file is blank.
+ */
+export async function getSdkSuppressionsSdkNames(
+  appClient: Octokit,
+  pr: PullRequestContext,
+  prChangeFiles: string
+): Promise<SdkName[]> {
+  
+  const filesChangedPaths = prChangeFiles.split(' ');
+  // Get suppression file list
+  let suppressionFileList = filterSuppressionList(filesChangedPaths);
+
+  let sdkNameList: SdkName[] = [];
+  // Get suppression file Content
+  if (suppressionFileList.length > 0) {
+    for (const suppressionFile of suppressionFileList) {
+      logSuppressionFileInfo(
+        {
+          owner: pr.base.owner,
+          repo: pr.base.repo,
+          ref: pr.base.ref,
+          path: suppressionFile,
+        }
+      );
+      let baseSuppressionContent = await getSdkSuppressionsFileContent(
+        appClient,
+        pr.base.owner,
+        pr.base.repo,
+        pr.base.ref,
+        suppressionFile
+      );
+      logSuppressionFileInfo(
+        {
+          owner: pr.head.owner,
+          repo: pr.head.repo,
+          ref: pr.head.ref,
+          path: suppressionFile,
+        }
+      );
+      const headSuppressionContent = await getSdkSuppressionsFileContent(
+        appClient,
+        pr.head.owner,
+        pr.head.repo,
+        pr.head.ref,
+        suppressionFile
+      );
+
+      // if the head suppression file is present but anything is wrong like schema error with it return
+      const validateSdkSuppressionsFileResult = validateSdkSuppressionsFile(headSuppressionContent).result;
+      if (!validateSdkSuppressionsFileResult) {
+        return [];
+      }
+      // if base suppression file does not exist, set it to an empty object but has correct schema
+      if (!baseSuppressionContent) {
+        baseSuppressionContent = { suppressions: {} };
+      }
+
+      console.log(
+        `updateSdkSuppressionsLabels: PR: ${pr.html_url} Will compare base suppressions content: ${JSON.stringify(
+          baseSuppressionContent
+        )} and head suppressions content: ${JSON.stringify(headSuppressionContent)} to get different SDK.`
+      );
+
+      sdkNameList = getSdkNamesWithChangedSuppressions(
+        headSuppressionContent as SdkSuppressionsYml,
+        baseSuppressionContent as SdkSuppressionsYml
+      );
+    }
+  }
+
+  return [...new Set(sdkNameList)];
+}
+
+async function getSdkSuppressionsFileContent(
+  appClient: Octokit,
```

Comment:
Talked to Mike and we would like to use the git history locally instead of calling GitHub. Try to see call 'git' command or 'simpleGit' to checkout to the corresponding commitSha locally to get the content.